giftless/storage/google_cloud.py

```diff
@@ -97,21 +103,84 @@ def _get_blob_path(self, prefix: str, oid: str) -> str:
             storage_prefix = self.path_prefix
         return os.path.join(storage_prefix, prefix, oid)
 
-    def _get_signed_url(self, prefix: str, oid: str, expires_in: int, filename: Optional[str] = None) -> str:
-        bucket = self.storage_client.get_bucket(self.bucket_name)
-        blob = bucket.blob(self._get_blob_path(prefix, oid))
-        token_expires = (datetime.now(tz=timezone.utc) +
-                         timedelta(seconds=expires_in))
-
-        extra_args = {}
-        if filename:
-            extra_args['content_disposition'] = f'attachment; filename="{filename}"'
-
-        url = blob.generate_signed_url(
-            response_disposition=extra_args,
-            access_token=self.api_key,
-            expiration=token_expires, version='v4')
-        return url  # type: ignore
+    def _get_signed_url(self, object_name: str,
```

Comment:
This is a very long method. Please break it down. It will make things a lot more readable. I haven't read it yet (because reading an 80 line long method is scary) but once I do I will make some in-line suggestions on breaking things down.

Reply:
Actually, I just now realized that this is our own implementation of URL signing for Google. 

Why are we implementing our own signing (and bringing in a lot of dependencies and complexity in the process) instead of using Google Cloud's own implementation (e.g. in https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/storage/cloud-client/storage_generate_upload_signed_url_v4.py)?

Reply:
This is what I tried initially and could not get it to work; so I reimplemented the signed_url. I gave it another try now; no success. What do you suggest? Use more time trying to make it to work, or go for this implementation?

Reply:
Do we know why it doesn't work? If you want we can spend some time trying to debug this together. I would really expect Google's code to work best or if it doesn't work in our specific case that we understand why. Implementing our own should be the last resort.

Reply:
We could try to pair and investigate, but the problem is that our work hours do not overlap that much; esp this week

Reply:
I have created a separate ticket to handle this (#38) and will resolve for now. If I have time I will try to take a look as well.